code/modules/tgui/tgui_window.dm

```diff
@@ -267,8 +298,6 @@
 		return
 	sent_assets |= list(asset)
 	. = asset.send(client)
-	if(!.)
```

Comment:
This fixes a bunch of crap. You may want to look at reverting a090357443f9440169ed3f935162116e20e87857 since it appears to have broken a bunch of this.